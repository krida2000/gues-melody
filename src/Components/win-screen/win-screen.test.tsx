import * as React from "react";
import * as renderer from "react-test-renderer";
import WinScreen from "./win-screen";
import {BrowserRouter} from "react-router-dom"
const noop = () => {
  // do nothing
};

describe(`Win screen correctly render`, () => {
  it(`if you really win`, () => {
    const tree = renderer
      .create(
        <WinScreen
          mistakes={0}
          maxMistakes={0}
          steep={10}
          replayButtonClickHandler={noop}
        />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`if you don't win`, () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <WinScreen
            mistakes={0}
            maxMistakes={0}
            steep={5}
            replayButtonClickHandler={noop}
          />
        </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
})
