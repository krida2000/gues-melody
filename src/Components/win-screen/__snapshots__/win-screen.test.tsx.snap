// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Win screen correctly render if you don't win 1`] = `null`;

exports[`Win screen correctly render if you really win 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Ура победа!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    Вы ответили на все вопросы, сделав всего 
    0
     шибок из допустимых 
    0
    !
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Попробовать ещё раз
  </button>
</section>
`;
